sys/linux/socket_netlink_route.txt

```diff
@@ -879,6 +877,32 @@ rtcanmsg {
 
 rtcanmsg_flags = CGW_FLAGS_CAN_ECHO, CGW_FLAGS_CAN_SRC_TSTAMP, CGW_FLAGS_CAN_IIF_TX_OK
 
+lwtunnel_encap_attr [
+	generic	array[int8]
```

Comment:
Instead of these `generic array[int8]` and `nlattr_tt[int16:14[0:150]` we need actual netlink policies that are valid for RTA_ENCAP.

You may see it's read here:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/ipv6/route.c#L4994
and assigned to cfg->fc_encap.

This cfg->fc_encap is later used in a call to fib_nh_common_init here:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/ipv6/route.c#L3509

fib_nh_common_init passes it to lwtunnel_build_state here:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/ipv4/fib_semantics.c#L593

lwtunnel_build_state passes it to build_state callback based on RTA_ENCAP_TYPE: 
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/core/lwtunnel.c#L127

These callbacks are registered with lwtunnel_encap_add_ops:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/core/lwtunnel.c#L78

which is called from these places:
https://elixir.bootlin.com/linux/v5.8-rc4/C/ident/lwtunnel_encap_add_ops

If we look at the bpf call:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/core/lwt_bpf.c#L655

We see it registers bpf_build_state callback, which uses bpf_nl_policy to parse the attributes:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/core/lwt_bpf.c#L383

So this is the policy we need to describe for LWTUNNEL_ENCAP_BPF:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/core/lwt_bpf.c#L363

And so on for other encap types.

Reply:
I added actual netlink policies that are valid for RTA_ENCAP. 
If you are happy with the changes for rpl_lwtunnel, I will go and add descriptions for other lwtunnel_encap types.
I will probably do it in the same PR.